source/common/stats/symbol_table_impl.h

```diff
@@ -356,6 +422,18 @@ class StatNameJoiner {
   std::unique_ptr<SymbolStorage> bytes_;
 };
 
+class StatNameTempStorage : public StatNameStorage {
+public:
+  StatNameTempStorage(absl::string_view name, SymbolTable& table)
```

Comment:
Same comment as above: Unclear why these three are separate constructors, maybe a note about the uses of each.